pkg/util/mount/mount_helper_common.go

```diff
@@ -95,9 +50,6 @@ func PathExists(path string) (bool, error) {
 		return true, nil
 	} else if os.IsNotExist(err) {
 		return false, nil
-	} else if IsCorruptedMnt(err) {
-		return true, err
-	} else {
-		return false, err
 	}
+	return false, err
```

Comment:
I did a scan of all places using mount.PathExists and think there are quite a few places having issues when using this to teardown volume mounts.
e.g., 
```
func (detacher *rbdDetacher) UnmountDevice(deviceMountPath string) error {
	if pathExists, pathErr := mount.PathExists(deviceMountPath); pathErr != nil {
		return fmt.Errorf("Error checking if path exists: %v", pathErr)
	} else if !pathExists {
		klog.Warningf("Warning: Unmount skipped because path does not exist: %v", deviceMountPath)
		return nil
	}
```
the problem of this case is that it immediately return if there is an error when calling PathExists. It loses the chance to clean up for corrupted mounts.

Another example is 
```

// UnmountDevice is part of the volume.Detacher interface.
func (d *flexVolumeDetacher) UnmountDevice(deviceMountPath string) error {

	pathExists, pathErr := mount.PathExists(deviceMountPath)
	if !pathExists {
		klog.Warningf("Warning: Unmount skipped because path does not exist: %v", deviceMountPath)
		return nil
	}
	if pathErr != nil && !mount.IsCorruptedMnt(pathErr) {
		return fmt.Errorf("Error checking path: %v", pathErr)
	}
```
In this case, it also loses the chance to clean up the corrupted mount after your change.

I think it is better to change all such places to use the similar way of checking in your CleanupMountPoint function

Reply:
Should we change all places to use CleanupMountPoint?